### PR TITLE
Suppress resolvedUsingTsExtension during loadModuleFromDirectory

### LINKED ISSUE #52182: Debug failure at resolveExternalModule after allowArbitraryExtensions
As of https://github.com/microsoft/TypeScript/commit/89e928e8b41368bb0888c6d399854c53a2dbf3d4 (#51435), the perf runner is Debug asserting: https://typescript.visualstudio.com/TypeScript/_build/results?buildId=142178&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b

```
>> /home/setup/vsts-agent/_work/7/s/built/local/tsc.js:109755
>>       throw e;
>>       ^
>> 
>> Error: Debug Failure.
>>     at resolveExternalModule (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:44740:35)
>>     at resolveExternalModuleNameWorker (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:44702:61)
>>     at resolveExternalModuleName (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:44699:12)
>>     at checkImportDeclaration (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:76954:35)
>>     at checkSourceElementWorker (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:77425:16)
>>     at checkSourceElement (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:77244:7)
>>     at forEach (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:37:22)
>>     at checkSourceFileWorker (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:77608:7)
>>     at checkSourceFile (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:77575:5)
>>     at checkSourceFileWithEagerDiagnostics (/home/setup/vsts-agent/_work/7/s/built/local/tsc.js:77669:5
```

This seems to be carried through to PRs too (like https://typescript.visualstudio.com/TypeScript/_build/results?buildId=142192&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=fb1961db-d514-5215-fe81-a42cdc309f86 😢)

cc @weswigham

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #52182

This is the fix suggested by @weswigham; the state prop could also be undefined and implicit to avoid setting false everywhere, but I am guessing this makes it slower somehow.

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot test this
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot run dt
@typescript-bot perf test this

**Review by weswigham — APPROVED** [review]
This is the only callsite where we construct a path and feed it into `nodeLoadModuleByRelativeName` where that path doesn't come from a user-supplied specifier, right? Right.

**Comment by jakebailey** [comment]
I think so? Interestingly, it's also the only place where we pass `considerPackageJson=false`, so maybe _that_ is the flag I should be using?

**Comment by andrewbranch** [comment]
I think the meaningful thing is whether the module name you’re currently trying to resolve actually came from an import module specifier in source code (this is the only time `resolvedUsingTsExtension` matters / should be set) or from a filename listed in a config file.

Actually, it’s a bit more nuanced than that. What matters is whether the file extension itself came from an import module specifier. The module name you’re resolving could be a combination of source text and path mapping config text constructed via a wildcard substitution, e.g.:

```
// ts
import "foo/bar.ts"
import "baz/main.ts"

// tsconfig
"paths": {
  "foo/*": ["./dist/*"],
  "baz/*.ts": ["./types/*.d.ts"]
}
```

Assuming both of these resolve, `resolvedUsingTsExtension` should be set on the former but not on the latter 😵. I should have thought to add tests like this.

**Comment by jakebailey** [comment]
I'm basically just a monkey at a typewriter on this one having never touched this code before... 😅 

Are you saying that this fix isn't complete/right and I should be fixing things up? (I'll add that case in and see what happens.)

File: tests/baselines/reference/issue52182.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//// [tests/cases/compiler/issue52182.ts] ////
```
Comment (jakebailey) [inline tests/baselines/reference/issue52182.js:1]: This test name is a placeholder I forgot to change; will modify.

(Honestly I wish we just used `issue#` file names so I didn't have to dream up names, but, there aren't any like that in the repo.)

**Comment by jakebailey** [comment]
I added what I think is the test you suggested? (Though, maybe not, given this bug is only in the node10 resolver? Maybe?)

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**